### ISSUE TITLE
Code Coloring: Python's new f-string feature is not fully recognized

### ISSUE THREAD

**Issue by aliyilmazz** [atom/atom#17213]
<!--

### Prerequisites

* [ ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Hello, I'm coding in python3.6 with Atom, and I discovered that python3.6's `f-string` feature is not fully recognized. Recall from f-string PEP documentation: 
```
F-strings provide a way to embed expressions inside string literals, using a minimal syntax. 
```

Quick Example:
```python3
>>> value = 80
>>> f'The value is {value}.'
'The value is 80.'
```

As you can see, we can use variables inside strings, which is quite awesome.
However, Atom does not recognize all types of variables and `color` them accordingly.
Here is the source of this issue:
![alt text](https://image.ibb.co/fm7fGx/Screenshot_from_2018_04_25_12_58_07.png)

`project.title` is recognized as a variable, and colored. all good. on the other hand, `args['sessionId']` is not recognized as a variable. 

In short, I believe that dictionary elements are not recognized and syntax-beautified within f-strings.

### Steps to Reproduce

1. Open a .py file in Atom
2. Paste this code into your editor:
```python3
>>> value = 80
>>> f'The value is {value}.'
'The value is 80.'
```
3. Enjoy
**Expected behavior:** [What you expect to happen]

**Actual behavior:** [What actually happens]

**Reproduces how often:** [What percentage of the time does it reproduce?]

### Versions

atom --version
Atom    : 1.24.1
Electron: 1.6.16
Chrome  : 56.0.2924.87
Node    : 7.4.0

I tested this on Ubuntu 16.04 

### Additional Information

Also there is a problem with git's `amend` checkbox. I observed that `amend` checkbox caches old amend stuff from old branches, which I will create a ticket soon! Happy hacking!

**Comment by aliyilmazz** [comment]
Is this intended?

![alt text](https://image.ibb.co/ca4PRx/Screenshot_from_2018_04_25_14_16_04.png)

**Comment by thomasjo** [comment]
> I'm sorry about the wrong-repository thing.

No problem, I was just letting you know — no harm done :smile: 

> updated my atom, rebooted my PC, dictionary elements work fine. However, I still can't get color on myvalue. Is this intended?

Based on my testing, the problem is that the `meta.interpolation` scope is not given a specific color in the `one-[dark/light]-syntax` themes, which is why it looks like a normal string. The reason the `request.args` expression is getting highlighted is that it's being recognized as a property expression, which is specifically treated in the syntax themes.

@simurai What do you think about differentiating the interpolation part of these types of strings in Python? In the case of JavaScript we're using different scopes though (`embedded`).

/cc @50Wliu

**Comment by thomasjo** [comment]
For future posterity, as of writing this, f-strings are not supported when using Tree-sitter parsers in Atom. The relevant issue can be found here: https://github.com/tree-sitter/tree-sitter-python/issues/26

**Comment by 50Wliu** [comment]
I don't recall off the top of my head why Python is using interpolation instead of embedded.  I'll have to check some other languages to see what they do.

**Comment by simurai** [comment]
> @thomasjo What do you think about differentiating the interpolation part of these types of strings in Python?

👍 Sure... I can add it. Unless @50Wliu wants to use `embedded` so it matches other languages?
